spikeinterface/toolkit/preprocessing/normalize_scale.py

```diff
@@ -7,15 +7,16 @@
 
 class ScaleRecordingSegment(BasePreprocessorSegment):
     # use by NormalizeByQuantileRecording/ScaleRecording/CenterRecording
-    def __init__(self, parent_recording_segment, gain, offset):
+    def __init__(self, parent_recording_segment, gain, offset, dtype):
         BasePreprocessorSegment.__init__(self, parent_recording_segment)
```

Comment:
you need to propagate the dtype to the base class with
 `BasePreprocessorSegment.__init__(self, parent_recording_segment, dtype=dtype)`